lib/puppet/defaults.rb

```diff
@@ -606,16 +606,16 @@ def self.default_vendormoduledir
       configuration files.  This timeout determines how quickly Puppet checks whether
       a file (such as manifests or puppet.conf) has changed on disk. The default will
       change in a future release to be 'unlimited', requiring a reload of the Puppet
-      service to pick up changes to its internal configuration. To reparse files
-      within an environment in Puppet Server please use the environment_cache endpoint",
+      service to pick up changes to its internal configuration. Currently we do not
+      accept a value of 'unlimited'. To reparse files within an environment in
+      Puppet Server please use the environment_cache endpoint",
       :hook => proc do |val|
-        if ![0, '15s'].include?(val)
+        unless [0, 15, '15s'].include?(val)
           Puppet.deprecation_warning(<<-WARNING)
 Fine grained control of filetimeouts is deprecated. In future
 releases this value will only determine if file content is cached.
 
-Valid values will be 0 (never cache) and 'unlimited' (always cache).
-With a default of 'unlimited'.
+Valid values are 0 (never cache) and 15 (15 second minimum wait time).
```

Comment:
I updated the warning and the main description to clarify valid values currently, and the use of `unlimited` in the future.